packages/rest/src/body-parsers/body-parser.ts

```diff
@@ -50,18 +53,60 @@ export class RequestBodyParser {
         new RawBodyParser(options),
       ];
     } else {
-      this.parsers = parsers;
+      this.parsers = sortParsers(parsers);
+    }
+    if (debug.enabled) {
+      debug('Body parsers: ', this.parsers.map(p => p.name));
     }
   }
 
   async loadRequestBodyIfNeeded(
     operationSpec: OperationObject,
     request: Request,
   ): Promise<RequestBody> {
+    const {requestBody, customParser} = await this._matchRequestBodySpec(
+      operationSpec,
+      request,
+    );
+    if (!operationSpec.requestBody) return requestBody;
+    const matchedMediaType = requestBody.mediaType!;
+    try {
+      if (customParser) {
+        // Invoke the custom parser
+        const body = await this._invokeCustomParser(customParser, request);
+        if (body !== undefined) return Object.assign(requestBody, body);
```

Comment:
Please see my earlier comment on this topic.

Why are you falling back to iterate through `this.parsers` when `_invokeCustomParser` returns `undefined`? I find it very confusing.

Reply:
Fixed